### ISSUE TITLE
Cycle in toString of RecoveryCodeEntity and ApplicationEntity

### ISSUE THREAD

**Issue by banterCZ** [wultra/powerauth-server#1689]
Should be fixed to `1.6.x` and `1.7.x`

```log
java.lang.StackOverflowError: null
	at java.base/java.lang.Object.toString(Unknown Source)
	at java.base/java.lang.String.valueOf(Unknown Source)
	at java.base/java.lang.StringBuilder.append(Unknown Source)
	at java.base/java.util.AbstractCollection.toString(Unknown Source)
	at org.hibernate.collection.spi.PersistentBag.toString(PersistentBag.java:590)
	at java.base/java.lang.StringConcatHelper.stringOf(Unknown Source)
	at io.getlime.security.powerauth.app.server.database.model.entity.ApplicationEntity.toString(ApplicationEntity.java:170)
	at io.getlime.security.powerauth.app.server.database.model.entity.RecoveryCodeEntity.toString(RecoveryCodeEntity.java:387)
	at java.base/java.lang.String.valueOf(Unknown Source)
	at java.base/java.lang.StringBuilder.append(Unknown Source)
	at java.base/java.util.AbstractCollection.toString(Unknown Source)
	at org.hibernate.collection.spi.PersistentBag.toString(PersistentBag.java:590)
	at java.base/java.lang.StringConcatHelper.stringOf(Unknown Source)
	at io.getlime.security.powerauth.app.server.database.model.entity.ApplicationEntity.toString(ApplicationEntity.java:170)
	at io.getlime.security.powerauth.app.server.database.model.entity.RecoveryCodeEntity.toString(RecoveryCodeEntity.java:387)
	at java.base/java.lang.String.valueOf(Unknown Source)
	at java.base/java.lang.StringBuilder.append(Unknown Source)
	at java.base/java.util.AbstractCollection.toString(Unknown Source)
	at org.hibernate.collection.spi.PersistentBag.toString(PersistentBag.java:590)
	at java.base/java.lang.StringConcatHelper.stringOf(Unknown Source)
	at io.getlime.security.powerauth.app.server.database.model.entity.ApplicationEntity.toString(ApplicationEntity.java:170)
	at io.getlime.security.powerauth.app.server.database.model.entity.RecoveryCodeEntity.toString(RecoveryCodeEntity.java:387)
	at java.base/java.lang.String.valueOf(Unknown Source)
	at java.base/java.lang.StringBuilder.append(Unknown Source)
	at java.base/java.util.AbstractCollection.toString(Unknown Source)
	at org.hibernate.collection.spi.PersistentBag.toString(PersistentBag.java:590)
	at java.base/java.lang.StringConcatHelper.stringOf(Unknown Source)
```

**Comment by zcgandcomp** [comment]
@banterCZ please confirm here.

**Comment by banterCZ** [comment]
Yes, it is fixed and also backported to 1.6 and 1.7 in #1691 and #1690
